### PR TITLE
Handle SIGTERM on OSX

### LINKED ISSUE #7394: AssemblyLoadContext.Unloading is not called when TERM signal sent to application
Based on [#2688](https://github.com/dotnet/coreclr/issues/2688) and [#5205](https://github.com/dotnet/corefx/issues/5205), I would expect the default AssemblyLoadContext's Unloading event to fire when the application receives a SIGTERM.  However, the event handler doesn't seem to ever be called.

Code to reproduce:

```
public class Program
{
private static void unloadTask(AssemblyLoadContext obj)
{
    Console.WriteLine("Unloading");
}

    public static void Main(string[] args)
    {
        Console.WriteLine("Starting App");

        AssemblyLoadContext.Default.Unloading += unloadTask;

        try
        {
        Console.ReadLine();

        }
        catch (System.Exception)
        {
            Console.WriteLine("aborted?");
        }
    }

}
```

$dotnet run
$kill -TERM pid

output:
Starting App

expected output:
Starting App
Unloading

Same behavior with dotnet 1.0.1 on OSX and Windows 10

### PR DESCRIPTION
**Pull request by kouvel**
Fixes #7394

### CODE REVIEW

File: src/pal/src/exception/signal.cpp [hunk 1]
```diff
@@ -67,56 +70,67 @@ typedef void *siginfo_t;
...
+#if !HAVE_MACH_EXCEPTIONS
 static void sigill_handler(int code, siginfo_t *siginfo, void *context);
 static void sigfpe_handler(int code, siginfo_t *siginfo, void *context);
 static void sigsegv_handler(int code, siginfo_t *siginfo, void *context);
 static void sigtrap_handler(int code, siginfo_t *siginfo, void *context);
 static void sigbus_handler(int code, siginfo_t *siginfo, void *context);
 static void sigint_handler(int code, siginfo_t *siginfo, void *context);
 static void sigquit_handler(int code, siginfo_t *siginfo, void *context);
+#endif // !HAVE_MACH_EXCEPTIONS
 static void sigterm_handler(int code, siginfo_t *siginfo, void *context);
```
Comment (janvorli) [inline src/pal/src/exception/signal.cpp]: A nit - could you please move this line above all of the handler declarations? Then you can use single `#if !HAVE_MACH_EXCEPTIONS`

File: src/pal/src/exception/signal.cpp [hunk 2]
```diff
@@ -67,56 +70,67 @@ typedef void *siginfo_t;
...
 struct sigaction g_previous_sigtrap;
 struct sigaction g_previous_sigfpe;
 struct sigaction g_previous_sigbus;
 struct sigaction g_previous_sigsegv;
 struct sigaction g_previous_sigint;
 struct sigaction g_previous_sigquit;
+#endif // !HAVE_MACH_EXCEPTIONS
 struct sigaction g_previous_sigterm;
 
 static bool registered_sigterm_handler = false;
```
Comment (janvorli) [inline src/pal/src/exception/signal.cpp]: A nit - similar to what I've said above, could you please move the two variables above all the `struct sigaction xxxx` declarations above and use single `#if !HAVE_MACH_EXCEPTIONS`?

**Comment by kouvel** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by kouvel** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by kouvel** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test

**Comment by danmosemsft** [comment]
@dotnet-bot test OSX10.12 x64 Checked Build and Test  (network)

**Pull request merged**